pkg/webservice/handlers_test.go

```diff
@@ -175,3 +177,87 @@ func TestContainerHistory(t *testing.T) {
 	assert.Equal(t, contHist[0].TotalContainers, "2", "metric 1 should be 1 apps and was not")
 	assert.Equal(t, contHist[4].TotalContainers, "300", "metric 5 should be 300 apps and was not")
 }
+
+func TestQueryParamInAppsHandler(t *testing.T) {
+	const configDefault = `
+partitions:
+  - name: default
+    queues:
+      - name: root
+        queues:
+        - name: default
+`
+	rmID := "rm-123"
+	policyGroup := "default-policy-group"
+	clusterInfo := cache.NewClusterInfo()
+
+	configs.MockSchedulerConfigByData([]byte(configDefault))
+	if _, err := cache.SetClusterInfoFromConfigFile(clusterInfo, rmID, policyGroup); err != nil {
+		t.Errorf("Error when load clusterInfo from config %v", err)
+		return
+	}
+	assert.Equal(t, 1, len(clusterInfo.ListPartitions()))
+
+	// Check default partition
+	partitionName := "[" + rmID + "]default"
+	defaultPartition := clusterInfo.GetPartition(partitionName)
+	assert.Equal(t, 0, len(defaultPartition.GetApplications()))
+
+	// add a new app
+	appInfo := cache.CreateNewApplicationInfo("app-1", partitionName, "root.default")
+	err := cache.AddAppToPartition(appInfo, defaultPartition)
+	if err != nil {
+		t.Errorf("add application to partition should not have failed: %v", err)
+	}
+	assert.Equal(t, appInfo.GetApplicationState(), "New")
+	assert.Equal(t, 1, len(defaultPartition.GetApplications()))
+
+	NewWebApp(clusterInfo, nil)
+
+	// Passing "root.default" as filter return 1 application
+	req, errReq := http.NewRequest("GET", "/ws/v1/apps?queue=root.default", strings.NewReader(""))
+	assert.NilError(t, errReq, "App Handler request failed")
+	resp := &MockResponseWriter{}
+	var appsDao []*dao.ApplicationDAOInfo
+	getApplicationsInfo(resp, req)
+	err = json.Unmarshal(resp.outputBytes, &appsDao)
+	assert.NilError(t, err, "failed to unmarshal applications dao response from response body: %s", string(resp.outputBytes))
+	assert.Equal(t, len(appsDao), 1)
+
+	// Passing "root.q1" as filter return 0 application as there is no app running in "root.q1" queue
+	req1, err1 := http.NewRequest("GET", "/ws/v1/apps?queue=root.q1", strings.NewReader(""))
```

Comment:
You can reuse variables: `resp`, `req` and `errReq`, no need to create req1..req4
You can even reuse `err` instead of `errReq` by declaring `var req *http.Request` before line 218